docs/articles/vault.md

```diff
@@ -1,43 +1,194 @@
+on the same port, you can specify a different port number (i.e. `8201`) for the
+agent by adding the highlighted lines:
 
 ```yaml
-global:
-    testWorkflows:
-        globalTemplate:
-            enabled: true
-            spec:
-                pod:
-                    annotations:
-                        vault.hashicorp.com/agent-inject-secret-foo: database/roles/app
-                        vault.hashicorp.com/agent-inject-secret-bar: consul/creds/app
-                        vault.hashicorp.com/role: "app"
+apiVersion: testworkflows.testkube.io/v1
+kind: TestWorkflow
+metadata:
+  name: check-vault-secret-injection
+spec:
+  pod:
+    serviceAccountName: test-vault
+    labels:
+      app: workflow-vault
+    annotations:
+      vault.hashicorp.com/agent-inject-secret-database-config.txt: internal/data/database/config
+      vault.hashicorp.com/role: internal-app
+  use:
+    - name: 'vault-secret-injection'
+  # highlight-start
+      config:
+        port: "8201"
+  # highlight-end
+  container:
+    image: alpine:latest
+  steps:
+    - name: Check secret injection
+      condition: always
+      run:
+        shell: |
+          if [ -f /vault/secrets/database-config.txt ]; then
+            echo "Secret file found."
+          else
+            echo "Secret file not found."
+            exit 1
+          fi
+
 ```
 
-For the other executors, you can set these annotations globally with:
+#### Istio compatibility
+
+In setups, utilizing a combination of Vault and Istio we recommend explicitly
+excluding outgoing traffic to Vault's agent from being routed through Istio's
+proxy. Otherwise, the workload could be put in a dead-locked state where one
+init container is waiting on another which cannot start till the latter
+completes.
+
+:::info
 
-Chart `testkube`:
+To read more about Istio's compatibility with Testkube please read the following
+[guide](./istio).
+
+:::
+
+This can be configured by adding the following line to your workflow template:
 
 ```yaml
-testkube-api:
-    jobPodAnnotations:
-        vault.hashicorp.com/agent-inject-secret-foo: database/roles/app
-        vault.hashicorp.com/agent-inject-secret-bar: consul/creds/app
-        vault.hashicorp.com/role: "app"
-```
\ No newline at end of file
+apiVersion: testworkflows.testkube.io/v1
+kind: TestWorkflowTemplate
+metadata:
+  name: vault-secret-injection
+  namespace: testkube-agent-poznan
+spec:
+  config:
+    port:
+      type: string
+      default: "8200"
+  pod:
+    annotations:
+      vault.hashicorp.com/agent-inject: "true"
+      vault.hashicorp.com/agent-init-first: "true"
+      vault.hashicorp.com/agent-enable-quit: "true"
+      vault.hashicorp.com/agent-cache-listener-port: "{{ config.port }}"
+      # highlight-next-line
+      traffic.sidecar.istio.io/excludeOutboundPorts: "{{ config.port }}"
+  after:
+  - name: 'Send quit signal to Vault agent'
+    container:
+      image: busybox
```

Comment:
Is that required?

Reply:
I think this one is required to make the `wget` request to quit the agent.

Reply:
We do have `wget` and other Busybox binaries available for all the images automatically